src/Swarm/Game/Value.hs

```diff
@@ -87,6 +88,11 @@ data Value where
   VRef :: Int -> Value
   deriving (Eq, Show, Generic, FromJSON, ToJSON)
 
+-- | Ensure that a value is not wrapped in 'VResult'.
+stripVResult :: Value -> Value
+stripVResult (VResult v _) = v
```

Comment:
Just to make sure, there is no way a `VResult` could contain a `VResult`, right?

Reply:
Uhh... well, it's not *supposed* to, but it's a good question.  We might as well make `stripVResult` recursive, just in case.